returnn/datasets/util/feature_extraction.py

```diff
@@ -467,3 +471,22 @@ def _get_random_permuted_audio(audio, sample_rate, opts, random_state):
     audio = librosa.effects.pitch_shift(audio, sr=sample_rate, n_steps=n_steps)
   opts.assert_all_read()
   return audio
+
+
+def _get_f0_values(audio, sample_rate, step_len=0.010, fmin=0, fmax=None):
+  """
+  Calls pyworld dio and stonemask to retrieve f_0 values and replaces all NaN positions(silence) with 0
+  :param numpy.ndarray audio: raw time signal
+  :param int sample_rate: e.g. 22050
+  :param float step_len: in seconds
+  :param int fmin: minimum frequency covered by mel filters
+  :param int|None fmax: maximum frequency covered by mel filters
+  :rtype: numpy.ndarray
+  :return: Pitch features for audio signal
+  """
+  import pyworld as pw  # noqa
```

Comment:
Is this really not possible via librosa?

Reply:
https://librosa.org/doc/main/generated/librosa.pyin.html ?

Reply:
Ah I just read your comment:

> Swapped away from librosa to pyworld because it is a lot faster.

This should be here in the code as a comment.

Reply:
For 10000 sequences `librosa.pyin` (which is what some papers which dont use pyworld use) took 1-2 days while this takes about 2 hours and is more far used in the papers for pitch prediction I read. 
E.g. EspNet2 uses [dio](https://github.com/espnet/espnet/blob/b008ac7d58e9ced1a9f8c89cc85ee69d9e9461ab/espnet2/tts/feats_extract/dio.py) with pyworld.